app/src/main/java/ar/com/wolox/android/training/ui/login/LoginPresenter.kt

```diff
@@ -0,0 +1,72 @@
+package ar.com.wolox.android.training.ui.login
+
+import android.content.SharedPreferences
+import android.util.Patterns
+import ar.com.wolox.android.training.model.IUserService
+import ar.com.wolox.android.training.utils.networkCallback
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices
+import javax.inject.Inject
+
+class LoginPresenter @Inject constructor(private val sharedPreferences: SharedPreferences,
+                                         private val vRetrofitServices: RetrofitServices)
+    : BasePresenter<ILoginView>() {
+
+    val userEmailKey = "UserEmail"
```

Comment:
This is not a contant value , you should try to create it into a companion object like "private const val userEmailKey = "USER_EMAIL""

Reply:
Solved in next PR update.